File: Backend/src/models/user_groups.py
# Backend/src/models/user_groups.py
# Many to Many relationship between users and groups

# Author: Valentin Haas, 2024

# Library imports
from sqlalchemy import Column, Integer
from sqlalchemy.ext.declarative import declarative_base

# Constants
BASE = declarative_base()


class UserGroupsTable(BASE):
    """Model to track the relationship between users and groups."""

    __tablename__ = "UserGroups"

    User = Column(Integer, primary_key=True, foreign_key="Users.Id")
    """The user in the relationship."""

    Group = Column(Integer, primary_key=True, foreign_key="Groups.Id")
    """The group in the relationship."""

File: Backend/src/models/usergroups.go
// Backend/src/models/usergroups.go
// Many to Many relationship between users and groups

// Author: Valentin Haas, 2025
package models

import "time"

// UserGroup tracks the relationship between users and groups.
type UserGroup struct {
	User      User      // Referenced User Object
	UserID    uint      `gorm:"primaryKey;not null;foreignKey:UserID"` // The user in the relationship.
	Group     Group     // Referenced Group Object
	GroupID   uint      `gorm:"primaryKey;not null;foreignKey:GroupID"` // The group in the relationship.
	CreatedAt time.Time // Time the model was created. Auto Populated by Gorm.
	UpdatedAt time.Time // Time the model was updated. Auto Populated by Gorm.
	DeletedAt time.Time // Time the model was deleted. Auto Populated by Gorm.
}
